flink/v1.14/flink/src/main/java/com/netease/arctic/flink/lookup/KVTableFactory.java

```diff
@@ -46,6 +48,12 @@ public KVTable<RowData> create(
       Predicate<RowData> rowDataPredicate) {
     Set<String> joinKeySet = new HashSet<>(joinKeys);
     Set<String> primaryKeySet = new HashSet<>(primaryKeys);
+    // keep the primary keys order with projected schema fields.
+    primaryKeys = projectSchema.asStruct().fields().stream()
```

Comment:
Should the order of `joinKeys` be adjust too here?

Reply:
I add a test about this case.